hops/featurestore.py

```diff
@@ -2369,89 +2369,160 @@ def disable_featuregroup_online(featuregroup_name, featuregroup_version=1, featu
+    """
+
+    return core._do_get_tags(name, featurestore, version, constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_RESOURCE)
+
+
+def remove_featuregroup_tags(name, version=1, featurestore=None):
     """
-    if not keys:
-        return core._do_get_metadata(featuregroup_name=featuregroup_name, name=None, featurestore=featurestore, featuregroup_version=featuregroup_version)
+    Removes all tags attached to a feature group
 
-    if type(keys) is not list:
-        raise ValueError("keys should be a list")
+    Example usage:
 
-    vals = {}
-    for k in keys:
-        vals.update(core._do_get_metadata(featuregroup_name, k, featurestore, featuregroup_version))
-    return vals
+    >>> # The API will default to the project's feature store
+    >>> featurestore.remove_featuregroup_tags(featuregroup_name)
+    >>> # You can also explicitly override the default arguments:
+    >>> featurestore.remove_featuregroup_tags(featuregroup_name, version=1, featurestore=featurestore)
 
+    Args:
+        :name: name of the featuregroup
+        :version: version of the featuregroup
+        :featurestore: the featurestore that the featuregroup belongs to
 
-def remove_metadata(featuregroup_name, keys, featuregroup_version=1, featurestore=None):
+    Returns:
+        None
     """
-    Removes the custom metadata attached to a feature group
+
+    core._do_remove_tags(name, featurestore, version, constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_RESOURCE)
+
+
+def set_training_dataset_tags(name, tags, version=1, featurestore=None):
+    """
+    Attach tags to a training dataset
 
     Example usage:
 
     >>> # The API will default to the project's feature store
-    >>> metadata = featurestore.remove_metadata(featuregroup_name, ["attr1", "attr2"])
+    >>> featurestore.set_training_dataset_tags(training_dataset_name, {"attr1" : "val1", "attr2" : "val2"})
     >>> # You can also explicitly override the default arguments:
-    >>> metadata = featurestore.remove_metadata(featuregroup_name, ["attr1", "attr2"], featuregroup_version=1, featurestore=featurestore)
+    >>> featurestore.set_training_dataset_tags(training_dataset_name, {"attr1" : "val1", "attr2" : "val2"}, version=1, featurestore=featurestore)
 
     Args:
-        :featuregroup_name: name of the featuregroup
-        :keys: array of attribute names to be deleted from the featuregroup associated metadata
-        :featuregroup_version: version of the featuregroup
+        :name: name of the training dataset
+        :tags: a dictionary of attributes to attach to the training dataset
+        :version: version of the training dataset
+        :featurestore: the featurestore that the training dataset belongs to
+
+    Returns:
+        None
+    """
+    if type(tags) is not dict:
+        raise ValueError("tags should be a dictionary")
+
+    core._do_add_tags(name, tags, featurestore, version, constants.REST_CONFIG.HOPSWORKS_TRAININGDATASETS_RESOURCE)
```

Comment:
A suggestion would be to have one function `add_tags` which would accept the type, i.e featuregroup, training_datset, etc, as a parameter). Otherwise it's a bit duplicated code

Reply:
Initially I was thinking of that too but that would break how the current API looks like and make this function look off.

Reply:
For example we have methods like get_featuregroup_statistics and get_trainingdatasets_statistics

Reply:
Sounds good